<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ramp1dScan</class>
 <widget class="QDialog" name="ramp1dScan">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>670</width>
    <height>524</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Ramp 1D Scan Generator</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <layout class="QGridLayout" name="gridLayout">
     <item row="2" column="0">
      <widget class="QPushButton" name="pushAdd">
       <property name="text">
        <string>Add</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Number of points:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QPushButton" name="pushSave">
       <property name="text">
        <string>Save Values</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QPushButton" name="pushRandomize">
       <property name="text">
        <string>Randomize</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="spinNumPoints"/>
     </item>
     <item row="2" column="1">
      <widget class="QPushButton" name="pushProgramValues">
       <property name="text">
        <string>Program Values</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0" colspan="2">
      <widget class="QTableWidget" name="tableScanPoints"/>
     </item>
     <item row="4" column="1">
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>Load Values</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ramp1dScan</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>506</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ramp1dScan</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>512</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinNumPoints</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ramp1dScan</receiver>
   <slot>handleNumPointsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>667</x>
     <y>444</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushAdd</sender>
   <signal>clicked()</signal>
   <receiver>ramp1dScan</receiver>
   <slot>handleAddClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>204</x>
     <y>414</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>472</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushProgramValues</sender>
   <signal>clicked()</signal>
   <receiver>ramp1dScan</receiver>
   <slot>handleProgramValues()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>612</x>
     <y>408</y>
    </hint>
    <hint type="destinationlabel">
     <x>666</x>
     <y>483</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushRandomize</sender>
   <signal>clicked()</signal>
   <receiver>ramp1dScan</receiver>
   <slot>handleRandomizeClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>448</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>451</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushSave</sender>
   <signal>clicked()</signal>
   <receiver>ramp1dScan</receiver>
   <slot>handleSaveClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>510</x>
     <y>445</y>
    </hint>
    <hint type="destinationlabel">
     <x>666</x>
     <y>402</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>ramp1dScan</receiver>
   <slot>handleLoadClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>572</x>
     <y>478</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>485</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>handleNumPointsChanged()</slot>
  <slot>handleAddClicked()</slot>
  <slot>handleProgramValues()</slot>
  <slot>handleRandomizeClicked()</slot>
  <slot>handleSaveClicked()</slot>
  <slot>handleLoadClicked()</slot>
 </slots>
</ui>
